### ISSUE TITLE
Import .bai files if they are present when importing bam (instead of reindexing)

### ISSUE THREAD

**Issue by bwlang** [galaxyproject/galaxy#6181]
Not a big deal for small bams, but when importing 100G bam files index generation can really slow a workflow.

**Comment by mvdbeek** [comment]
It'd probably already help if we can use multiple threads for indexing, which gives a linear speedup on my measly 4 core macbook. https://github.com/pysam-developers/pysam/pull/755 should help.

**Comment by mschatz** [comment]
Agree, would love to see some effort here as this is now extremely common to pass around data as index BAM/CRAM files
